pulley/src/interp.rs

```diff
@@ -1439,6 +1439,26 @@ impl OpVisitor for Interpreter<'_> {
         self.state[dst].set_i64(src.into());
         ControlFlow::Continue(())
     }
+
+    fn xdiv32_s(&mut self, operands: BinaryOperands<XReg>) -> ControlFlow<Done> {
+        let me = self.current_pc::<crate::XDiv32S>();
+        let a = self.state[operands.src1].get_i32();
+        let b = self.state[operands.src2].get_i32();
+        match a.checked_div(b) {
+            Some(result) => {
+                self.state[operands.dst].set_i32(result);
+                ControlFlow::Continue(())
+            }
+            None => ControlFlow::Break(self.done_trap(me)),
```

Comment:
Could we make `done_trap` generic over an instruction and then avoid the need to do `let me = self.current_pc::<...>();` and instead do

```rust
ControlFlow::Break(self.done_trap::<...>())
```

or even fold the `ControlFlow::Break` into the helper?

This both delays the computation until its demanded (which LLVM can probably do, but might be foiled if there are zero/overflow whatever checks hidden in there) and is a little more concise at usage/call sites.

Reply:
Excellent idea 👍